src/app/dim-ui/Sheet.tsx

```diff
@@ -112,110 +128,53 @@ export default function Sheet({
-
+  const sheet = useRef<HTMLDivElement>(null);
   const sheetContents = useRef<HTMLDivElement | null>(null);
   const sheetContentsRefFn = useLockSheetContents(sheetContents);
+  const dragHandle = useRef<HTMLDivElement>(null);
 
-  useLayoutEffect(() => {
-    if (freezeInitialHeight && sheetContents.current && !frozenHeight) {
-      if (sheetContents.current.clientHeight > 0) {
-        setFrozenHeight(sheetContents.current.clientHeight);
-      } else {
-        setTimeout(() => {
-          sheetContents.current && setFrozenHeight(sheetContents.current.clientHeight);
-        }, 500);
-      }
-    }
-  }, [freezeInitialHeight, frozenHeight]);
-
-  const sheet = useRef<HTMLDivElement>(null);
-  const height = () => sheet.current?.clientHeight || 0;
+  const [frozenHeight, setFrozenHeight] = useState<number | undefined>(undefined);
+  const [disabled, setParentDisabled] = useDisableParent(forceDisabled);
 
-  /** When the sheet stops animating, if we were closing, fire the close callback. */
-  const onRest = useCallback(() => {
-    if (closing.current) {
-      onClose();
-    }
-  }, [onClose]);
-
-  /** This spring is controlled via setSpring, which doesn't trigger re-render. */
-  const [springProps, setSpring] = useSpring(() => ({
-    // Initially transition from offscreen to on
-    from: { transform: `translateY(${window.innerHeight}px)` },
-    to: { transform: `translateY(0px)` },
-    config: spring,
-    immediate: reducedMotion,
-    onRest,
-  }));
+  const reducedMotion = Boolean(useReducedMotion());
+  const animationControls = useAnimation();
+  const dragControls = useDragControls();
 
   /**
-   * Closing the sheet sets closing to true and starts an animation to close. We only fire the
-   * outer callback when the animation is done.
+   * Triggering close starts the animation. The onClose prop is called by the callback
+   * passed to the onAnimationComplete motion prop
    */
-  const handleClose = useCallback(
-    (e?: React.MouseEvent, dragDismiss?: boolean) => {
+  const triggerClose = useCallback(
+    (e?: React.MouseEvent) => {
       if (disabled) {
         return;
       }
       e?.preventDefault();
-      closing.current = true;
       // Animate offscreen
-      setSpring.start({
-        immediate: reducedMotion && !dragDismiss,
-        to: { transform: `translateY(${height()}px)` },
-      });
+      animationControls.start('close');
     },
-    [setSpring, disabled, reducedMotion]
+    [disabled, animationControls]
   );
 
   // Handle global escape key
-  useGlobalEscapeKey(handleClose);
-
-  // This handles all drag interaction. The callback is called without re-render.
-  const bindDrag = useDrag(({ event, active, movement, velocity, last, cancel }) => {
-    event?.stopPropagation();
-
-    if (isEventFromFirefoxScrollbar(event as any)) {
-      cancel();
-    }
-
-    // If we haven't enabled dragging, cancel the gesture
-    if (!last && cancel && !dragging.current) {
-      cancel();
-    }
```

Comment:
@bhollis since you brought up the `dragHandleUp` comment, I just figured I would call out this is the only "functionality" that has completely disappeared in this PR. I believe these are concerns that framer should be handling now but probably still worth pointing it out incase I am wrong here.

Reply:
I think this was from the Compare drag implementation. As long as it works in Firefox...

Reply:
I shall install it and find out